src/components/PeopleTable.tsx

```diff
@@ -1,645 +1,187 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+import React, { useEffect, useState } from 'react';
+import { Person as PersonType } from '../types';
+import { Person } from './Person';
+import { useParams, useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+
+type Props = {
+  peopleData: PersonType[] | null;
+  sex: string;
+  nameSearch: string;
+  centuries: string[];
+  sort: string;
+  direction: string;
+};
+
+export const PeopleTable: React.FC<Props> = ({
+  peopleData,
+  sex,
+  nameSearch,
+  centuries,
+  sort,
+  direction,
+}) => {
+  const { personName } = useParams();
+  const [local, setLocal] = useState(peopleData);
+  const [searchParam, setSearchParam] = useSearchParams();
+
+  useEffect(() => {
+    const filteredData =
+      peopleData?.filter(el => {
+        // Fix for correct century calculation
+        const numberCentury = el.born ? Math.floor(el.born / 100) + 1 : '';
+        const matchesSex = sex ? el.sex === sex : true;
+        const matchesName = nameSearch
+          ? el.name.toLowerCase().includes(nameSearch.toLowerCase())
+          : true;
+        const matchCentury =
+          centuries.length > 0
+            ? centuries.includes(numberCentury.toString())
+            : true;
+
+        return matchesSex && matchesName && matchCentury;
+      }) || [];
+
+    setLocal(filteredData);
+  }, [sex, nameSearch, peopleData, centuries]);
+
+  useEffect(() => {
+    if (!peopleData) {
+      return;
+    }
+
+    const sortedData = [...peopleData].sort((a, b) => {
+      let compareValue = 0;
+
+      if (sort === 'name') {
+        compareValue = a.name.localeCompare(b.name);
+      } else if (sort === 'born') {
+        compareValue = (a.born || 0) - (b.born || 0);
+      } else if (sort === 'died') {
+        compareValue = (a.died || 0) - (b.died || 0);
+      } else if (sort === 'sex') {
+        compareValue = a.sex.localeCompare(b.sex);
+      }
+
+      return direction === 'desc' ? -compareValue : compareValue;
+    });
+
+    setLocal(sortedData);
```

Comment:
Ensure that the sorting logic is applied to the filtered data (`local`) rather than the original `peopleData`. This will ensure that the displayed data is both filtered and sorted correctly.